src/aoefit.jl

```diff
@@ -327,3 +327,183 @@ function fit_single_aoe_compton(h::Histogram, ps::NamedTuple; uncertainty::Bool=
+            σ = f_aoe_sigma(e, (pars.σA, pars.σB))
+
+            # fit peak
+            try
+                A, B = fit_single_aoe_compton_with_fixed_μ_and_σ(h, μ, σ, ps; uncertainty=uncertainty)
+                neg_log_likelihoods[i] = A
+            catch e
+                @warn "Error fitting band $(compton_bands[i]): $e"
+                continue
+            end
+        end
+        return sum(neg_log_likelihoods)
+    end
+    
+    # MLE
+    opt_r = optimize(f_loglike ∘ inverse(f_trafo), f_trafo(v_init), NelderMead(), Optim.Options(time_limit = 60, show_trace=false, iterations = 1000))
+
+    converged = Optim.converged(opt_r)
+    !converged && @warn "Fit did not converge"
+
+    v_ml = inverse(f_trafo)(Optim.minimizer(opt_r))
+
+    if uncertainty && converged
+
+        f_loglike_array = p -> f_loglike((μA = p[1], μB = p[2], σA = p[3], σB = p[4]))
+
+        # Calculate the Hessian matrix using ForwardDiff
+        H = ForwardDiff.hessian(f_loglike_array, tuple_to_array(v_ml))
+
+        param_covariance = nothing
+        if !all(isfinite.(H))
+            @warn "Hessian matrix is not finite"
+            param_covariance = zeros(length(v_ml), length(v_ml))
+        else
+            # Calculate the parameter covariance matrix
+            param_covariance = inv(H)
+        end
+        if ~isposdef(param_covariance)
+            param_covariance = nearestSPD(param_covariance)
+        end
+        # Extract the parameter uncertainties
+        v_ml_err = array_to_tuple(sqrt.(abs.(diag(param_covariance))), v_ml)
+
+        # TODO: correctly combine p-values and residuals of the individual fits to the combined fit result
+        # get p-value 
+        # pval, chi2, dof = p_value(f_aoe_compton, h, v_ml)
+        # calculate normalized residuals
+        # residuals, residuals_norm, _, bin_centers = get_residuals(f_aoe_compton, h, v_ml)
+
+        @debug "Best Fit values"
+        @debug "μA: $(v_ml.μA) ± $(v_ml_err.μA)"
+        @debug "μB: $(v_ml.μB) ± $(v_ml_err.μB)"
+        @debug "σA: $(v_ml.σA) ± $(v_ml_err.σA)"
+        @debug "σB: $(v_ml.σB) ± $(v_ml_err.σB)"
+
+        result = merge(NamedTuple{keys(v_ml)}([measurement(v_ml[k], v_ml_err[k]) for k in keys(v_ml)]...),
+                #(gof = (pvalue = pval, chi2 = chi2, dof = dof, covmat = param_covariance, 
+                #residuals = residuals, residuals_norm = residuals_norm, bin_centers = bin_centers),)
+                )
+    else
+        @debug "Best Fit values"
+        @debug "μA: $(v_ml.μA)"
+        @debug "μB: $(v_ml.μB)"
+        @debug "σA: $(v_ml.σA)"
+        @debug "σB: $(v_ml.σB)"
+
+        result = merge(v_ml, )
+    end
+
+    # report = (
+    #         v = v_ml,
+    #         h = h,
+    #         f_fit = x -> Base.Fix2(f_aoe_compton, v_ml)(x),
+    #         f_sig = x -> Base.Fix2(f_aoe_sig, v_ml)(x),
+    #         f_bck = x -> Base.Fix2(f_aoe_bkg, v_ml)(x)
+    #     )
+
+    return result #, report
```

Comment:
We still need to define the format of `report` and maybe implemented/adjust a plot recipe for the fit report.